lib/livebook/session/data.ex

```diff
@@ -1454,6 +1467,17 @@ defmodule Livebook.Session.Data do
     end
   end
 
+  defp put_secret({data, _} = data_actions, secret) do
+    secrets =
+      if secret.label in get_in(data.secrets, [Access.all(), :label]) do
+        put_in(data.secrets, [Access.filter(&(&1.label == secret.label)), :value], secret.value)
+      else
+        data.secrets ++ [secret]
+      end
```

Comment:
Maybe we should just keep all secrets in alphabetic order? We can keep them as a map and then call Enum.sort before rendering them?

Reply:
Unless we plan to store any other information about a secret, this sounds great, and listing alphabetically is likely better for the end user!